model-optimizer/mo/utils/ir_reader/restore_graph.py

```diff
@@ -54,8 +65,18 @@ def save_restored_graph(graph: Graph, path: str, meta_data, name=None):
     if name is None:
         name = graph.name
 
-    precision = data_type_str_to_precision(graph.graph['cmd_params'].data_type)
-    assert precision in ['FP16', 'FP32'], 'Cannot define precision for restored model!'
+    if 'data_type' not in meta_data:
+        log.warning('Provided graph does not contain `data_type` parameter in `meta_info` section! Trying to '
```

Comment:
nit: `log.debug` because the previous solution based on `graph.graph['cmd_params'].data_type` is not quite correct

Reply:
Agree, updated.